src/profile-logic/call-tree.js

```diff
@@ -167,28 +168,45 @@ export class CallTree {
         icon:
           resourceType === resourceTypes.webhost
             ? extractFaviconFromLibname(libName)
-            : null,
+            : resourceType === resourceTypes.addon ? ExtensionIcon : null,
       };
       this._displayDataByIndex.set(callNodeIndex, displayData);
     }
     return displayData;
   }
 
   _getOriginAnnotation(funcIndex: IndexIntoFuncTable): string {
+    const resourceIndex = this._funcTable.resource[funcIndex];
+    const resourceNameIndex = this._resourceTable.name[resourceIndex];
+
+    let origin;
+    if (resourceNameIndex !== undefined) {
```

Comment:
I'm not sure about this. I didn't change this part of the logic, so if it's wrong now, it was also wrong before.

Reply:
I believe that fetching from an array can always be undefined if we have a bad resourceIndex -- Flow doesn't check for this.

On a side note I noticed a case where it could even be `null`, although this should happen very very rarely:
https://github.com/devtools-html/perf.html/blob/fbd739f3dcc762a2de34c92233fee117640e9cdf/src/profile-logic/old-cleopatra-profile-format.js#L273-L276

Reply:
@kmaglione if it was already that way feel free to ignore it and I'll file an issue on our side.